src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -195,11 +196,12 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
 
     /**
      * Returns an unmodifiable view of the list of {@code Record} backed by the internal list of
-     * {@code versionedAddressBook}
+     * {@code versionedAddressBook}. List of {@code Record} returned is sorted by Date, in
+     * descending order.
      */
     @Override
     public ObservableList<Record> getFilteredRecordList() {
-        return filteredRecords;
+        return filteredRecords.sorted(Collections.reverseOrder());
```

Comment:
Looks simple enough! We might need to consider the priority queue alternative when we perform system testing with a larger dataset to see if there is any obvious performance differences. Otherwise, good job man!

Reply:
Yes agreed, and thanks!!